sdk/storage/azure-storage-files-datalake/README.md

```diff
@@ -0,0 +1,199 @@
+- Management is easier because you can organize and manipulate files through directories and subdirectories.
+- Security is enforceable because you can define POSIX permissions on directories or individual files.
+- Cost effectiveness is made possible as Data Lake Storage Gen2 is built on top of the low-cost Azure Blob storage. The additional features further lower the total cost of ownership for running big data analytics on Azure.
+
+Data Lake Storage Gen2 offers two types of resources:
+
+- The _filesystem_ used via 'DataLakeFileSystemClient'
+- The _path_ used via 'DataLakeFileClient' or 'DataLakeDirectoryClient'
+
+|ADLS Gen2 	                | Blob       |
+| --------------------------| ---------- |
+|Filesystem                 | Container  | 
+|Path (File or Directory)   | Blob       |
+
+Note: This client library does not support hierarchical namespace (HNS) disabled storage accounts.
+
+Learn more about options for authentication (including Connection Strings, Shared Key, Shared Key Signatures, Active Directory, and anonymous public access) in our [samples](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/storage/azure-storage-files-datalake/samples).
+
+### Thread safety
+
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/cpp_introduction.html#thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+
+Client Options | [Accessing the response](https://github.com/Azure/azure-sdk-for-cpp#response-t-model-types) | [Long-running operations](https://github.com/Azure/azure-sdk-for-cpp#long-running-operations) | Handling failures
+
+## Examples
+
+### Appending Data to a DataLake File
+
+```C++
+const std::string fileSystemName = "sample-filesystem";
+const std::string directoryName = "sample-directory";
+const std::string fileName = "sample-file";
+
+auto sharedKeyCredential = std::make_shared<StorageSharedKeyCredential>(accountName, accountKey);
+
+// Create DataLakeServiceClient using StorageSharedKeyCredentials
+DataLakeServiceClient serviceClient = DataLakeServiceClient(serviceUrl, sharedKeyCredential);
+
+// Get a reference to a filesystem named "sample-filesystem" and then create it
+DataLakeFileSystemClient filesystemClient = serviceClient.GetFileSystemClient(fileSystemName);
+filesystemClient.CreateIfNotExists();
+
+// Create a DataLake Directory
+DataLakeDirectoryClient directoryClient = filesystemClient.GetDirectoryClient(directoryName);
+directoryClient.CreateIfNotExists();
+
+// Create a DataLake File using a DataLake Directory
+DataLakeFileClient fileClient = directoryClient.GetFileClient(fileName);
+fileClient.CreateIfNotExists();
+
+// Append data to the DataLake File
+FileBodyStream fileStream(localFilePath);
```

Comment:
Where is `FileBodyStream` defined? In which namespace?
We should probably add namespace qualifier if it's not in Datalake namespace